lecli/log_api.py

```diff
@@ -0,0 +1,174 @@
+"""
+Log API module.
+"""
+import json
+import requests
+
+from lecli import api_utils
+from lecli import response_utils
+
+def _url():
+    """
+    Get rest query url of log resource id.
+    """
+    return 'https://rest.logentries.com/management/logs'
+
+
+def handle_get_log_response(response):
+    """
+    Handle get log response.
+    """
+    if response_utils.response_error(response):
+        exit(1)
+    elif response.status_code == 200:
+        if 'log' in response.json():
+            print json.dumps(response.json()['log'], indent=4, sort_keys=True)
+        elif 'logs' in response.json():
+            print json.dumps(response.json()['logs'], indent=4, sort_keys=True)
+
+
+def get_logs():
+    """
+    Get logs associated with the user.
+    """
+    headers = api_utils.generate_headers('ro')
+    try:
+        response = requests.request('GET', _url(), headers=headers)
+        handle_get_log_response(response)
+    except requests.exceptions.RequestException as error:
+        print error
+        exit(1)
+
+
+def get_log(log_id):
+    """
+    Get a specific log.
+    """
+    headers = api_utils.generate_headers('ro')
+    try:
+        response = requests.get(_url() + "/" + str(log_id),
+                                headers=headers)
+        handle_get_log_response(response)
+    except requests.exceptions.RequestException as error:
+        print error
+        exit(1)
+
+
+def create_log(logname, params):
+    """
+    Add a new log to the current account.
+    If a filename is given, load the contents of the file
+    as json parameters for the request.
+    If a name is given, create a new log with the given name
+    """
+
+    if params is not None:
+        request_params = params
+    else:
+        request_params = {
+            'log': {
+                'name': str(logname)
```

Comment:
@udangel-r7 I agree, i think we may even have a bug around encoding strings like this. look at: https://github.com/rapid7/lecli/issues/31